crm_deduplicate_filter/wizards/partner_merge.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    exclude_is_company = fields.Boolean("'Is a company?' field selected")
+    exclude_not_parent = fields.Boolean("Parent company not set")
+    exclude_parent = fields.Boolean("Parent company set (Contacts)")
+
+    @api.multi
+    def _process_query(self, query):
+        if any([self.exclude_is_company, self.exclude_not_parent,
```

Comment:
This is only for performance. Avoid decomposing the query if there are no filters.